omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,232 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.idgen.rest.resource;
+
+import org.openmrs.api.context.Context;
+import org.openmrs.Location;
+import org.openmrs.PatientIdentifierType;
+import org.openmrs.module.idgen.IdentifierSource;
+import org.openmrs.module.idgen.AutoGenerationOption;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.module.idgen.web.controller.IdgenRestController;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Resource(name = RestConstants.VERSION_1 + IdgenRestController.IDGEN_NAMESPACE
+        + "/autogenerationoption", supportedClass = AutoGenerationOption.class, supportedOpenmrsVersions = { "1.9.*",
+                "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*" })
+public class AutoGenerationOptionResource extends DelegatingCrudResource<AutoGenerationOption> {
+	
+	IdentifierSourceService identifierSourceService = Context.getService(IdentifierSourceService.class);
```

Comment:
Why are you assigning this from here?

Reply:
the line Context.getService(IdentifierSourceService.class) was used in line 224,212,207,202,181,137 so I assigned it a variable identifierSourceService which is more compact, so in those lines every time identifierSourceService is used it represents Context.getService(IdentifierSourceService.class)

Reply:
Have you tried to compile and test the module to see the effects of that assignment?

Reply:
Yes  I have,  it works as it is expected,  I've attached the module's omod on the ticket

Reply:
Which rest call did you make to test it out? Can i see the results you got?

Reply:
all of them GET, POST, PUT, DELETE,  attached the screenshot results on the ticket.